src/components/Match/VisionPage.jsx

```diff
@@ -0,0 +1,231 @@
+import React from 'react';
+import {
+  Row,
+  Col,
+} from 'react-flexbox-grid';
+import { formatSeconds } from 'utility';
+import { Fixed } from 'utility/components';
+
+import Paper from 'material-ui/Paper';
+import Button from 'material-ui/RaisedButton';
+import Slider from 'material-ui/Slider';
+import Avatar from 'material-ui/Avatar';
+
+import Heading from 'components/Heading';
+import VisionMap from './VisionMap' ;
+import WardLog from './WardLog';
+import _ from 'lodash/fp';
+import Perf from 'react-addons-perf';
+import strings from 'lang';
+import styles from './Match.css';
+import { heroTd } from './matchColumns';
+import getMuiTheme from 'material-ui/styles/getMuiTheme';
+import {
+  grey800 as filterOff,
+  blueGrey700 as filterOn
+} from 'material-ui/styles/colors';
+
+window.Perf = Perf;
+
+const SliderTicks = (props) => (
+  <div {...props} >
+    {props.ticks.map((tick) => {
+       const percent = 100*(tick-props.min)/(props.max-props.min);
+       const cls = [styles['slider-tick']];
+       if (tick <= props.value)
+         cls.push(styles['active']);
+
+       return <a key={tick} onClick={() => props.onTickClick(tick)} className={cls.join(' ')} style={{left: percent + '%'}}>{formatSeconds(tick)}</a>
+     })}
+  </div>
+);
+
+
+class PlayerFilter extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.getObserverCount = () => this.props.player.obs_log.length;
```

Comment:
i'd advise making these functions outside the class, and take props as a parameter.